Pythonwin/pywin/framework/scriptutils.py

```diff
@@ -439,7 +441,7 @@ def ImportFile():
     # meaning sys.modules can change as a side-effect of looking at
     # module.__file__ - so we must take a copy (ie, list(items()))
     for key, mod in sys.modules.items():
-        if getattr(mod, "__file__", None):
+        if hasattr(mod, "__file__") and mod.__file__:
```

Comment:
how is this an improvement?

Reply:
Since dropping Python 3.7, this can be rewritten to use getattr and still be statically safe

Reply:
I have no idea what "statically safe" means for Python. In what specific scenarios would the old code fail and the new code not?

Reply:
I just mean that a tool can know statically (without executing any source code) that a piece of code is "sound" (in this case that `mod.__file__` exists and is truthy).

That being said, a more advanced type checker absolutely *could* understand `getattr`/`hasattr`. *And* this current patch/changeset is not enough to completely re-enable pyright's `reportCallIssue` to prevent possible real issues elsewhere. So I wouldn't even add a `# pyright: ignore[reportCallIssue]`.
So I'll just revert this line here.